apps/dashboard/src/hooks/use-update-workflow.ts

```diff
@@ -1,22 +1,17 @@
-import { useMutation } from '@tanstack/react-query';
-import type { UpdateWorkflowDto, WorkflowResponseDto } from '@novu/shared';
 import { updateWorkflow } from '@/api/workflows';
+import type { WorkflowResponseDto } from '@novu/shared';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
 
-export const useUpdateWorkflow = ({
-  onSuccess,
-  onError,
-}: { onSuccess?: (data: WorkflowResponseDto) => void; onError?: (error: unknown) => void } = {}) => {
-  const { mutateAsync, isPending, error, data } = useMutation({
-    mutationFn: async ({ id, workflow }: { id: string; workflow: UpdateWorkflowDto }) =>
-      updateWorkflow({ id, workflow }),
-    onSuccess,
-    onError,
+export const useUpdateWorkflow = (
+  options?: UseMutationOptions<WorkflowResponseDto, unknown, Parameters<typeof updateWorkflow>[0]>
```

Comment:
IMO we should not expose the react-query options as we will be tied by it's interface, but rather expose ours with the needed functionality.

Reply:
Ideally, I wouldn't create a `useUpdateWorkflow` hook in general. If we create one hook per mutation/query, we are already tied by its interface. I don't see any difference.

Reply:
the difference is that we will be free to change the implementation of the hook to whatever we want if we decide